### PR TITLE
[DOCFIX] Remove ALLUXIO_RELEASED_VERSION from docs/_config.yml

### PR DESCRIPTION
**Pull request by Xenorith**
Update references to ALLUXIO_RELEASED_VERSION to be
ALLUXIO_VERSION_STRING
Remove the even more obscure ALLUXIO_MASTER_VERSION_SHORT
Update ALLUXIO_MAJOR_VERSION on master to be "edge"

The concept of "most recently release version" changes as patches are
released, but this update is not reflected outside of the branch where
the patch was cut from
Instead of trying to update ALLUXIO_RELEASED_VERSION across multiple
branches, use ALLUXIO_VERSION_STRING which will be correct in all
branches except for master.
In the case of master, which is currently at 2.2.0-SNAPSHOT, it is
likely there will be invalid references (ex.
<version>2.2.0-SNAPSHOT</version> in the maven dependency declaration).
But I find this is to be more acceptable since it is a development
branch.

In summary, this PR introduces inconsistencies in the master branch but
corrects and simplifies version strings across all other release
branches.

pr-link: Alluxio/alluxio#10831
change-id: cid-07fa69cbfa91d587664ac1732917fd57ed28bdce

### CODE REVIEW

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please
